### PR TITLE
Remove extraArgs from v2plus log

### PR DESCRIPTION
**Pull request by vreff**
Remove extraArgs from fulfillment log and make requestCommitment memory as opposed to calldata
Saves ~2.2k gas

### CODE REVIEW

**Comment by jinhoonbang** [comment]
ExtraArgs in fulfillment log is nice to have because it can be used in benthos to populate `link_spent` and `native_spent` fields in consumers table for o11y. Optimizing gas seems to be higher priority though. This means we will need to calculate `link_spent` and `native_spent` by joining fulfillments and requests table and summing payments in the front end.

**Review by jinhoonbang — APPROVED** [review]

**Comment by vreff** [comment]
> ExtraArgs in fulfillment log is nice to have because it can be used in benthos to populate `link_spent` and `native_spent` fields in consumers table for o11y. Optimizing gas seems to be higher priority though. This means we will need to calculate `link_spent` and `native_spent` by joining fulfillments and requests table and summing payments in the front end.

Yep, we need to join the request with fulfillment off-chain to get the native true/false.

File: contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol [hunk 1]
```diff
@@ -348,7 +347,7 @@ contract VRFCoordinatorV2Plus is VRF, SubscriptionAPI {
 
   function getRandomnessFromProof(
     Proof calldata proof,
```
Comment (makramkd) [inline contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:349]: Would making this memory help or would compilation fail due to the slicing stuff?

Comment (vreff) [reply contracts/src/v0.8/dev/vrf/VRFCoordinatorV2Plus.sol:349]: I can check that out.

**Pull request merged**